### DISCUSSION TITLE
Relative links in snippets?

### DISCUSSION THREAD

**Discussion by feasgal** [squidfunk/mkdocs-material#5897] [Q&A] [answered]
So I'm using `pymdownx.snippets`, but my problem is that I want to put a link in the snippet, to one of the other pages in my docs set. Of course this doesn't work if the markdown files I use the snippet in don't all have the same relative path to the link destination.

I could begin the link url with a `/` to start all the way back at the domain and make it absolute (within the docs set), but the problem with that is that I generate two different docs sites from this set (one lite and one pro) and they have different `site_dir`s. So starting from the domain doesn't work. It would have to start from the `site_dir`.

So what I really want is for my link to basically be this pseudocode, but I haven't been able to find a way because the markdown source files can't 'see' the config values:

```
[My Destination](/{{ config.site_dir }}/rest/of/path.md)
```

Any ideas? I thought of somehow pulling `config.site_dir` into the template via my existing `main.html` override, but I couldn't quite wrap my head around how that would work.

**Comment by squidfunk** [comment]
You're at Markdown processing stage, not at template rendering which comes later, so moving anything inside `main.html` will likely not help. However, what you're aiming for is definitely achievable, either with the [macros plugin](https://mkdocs-macros-plugin.readthedocs.io/en/latest/), or with a [custom hook](https://www.mkdocs.org/user-guide/configuration/#hooks) that replaces a placeholder of your choosing before rendering the Markdown of the page. The latter solution, i.e., a simple string replacement, is likely faster, so depending on the size of your documentation it might be the desirable path to follow. Example:

``` py
def on_page_markdown(markdown, *, page, config, files):
    return markdown.replace("__DOCS_DIR__", config.docs_dir)
```

Note that I'm not sure if that's already sufficient to resolve the correct link location. Some fiddling might be necessary 😅 You should the be able to use this placeholder in your Markdown then:

``` md
[My Destination](__DOCS_DIR__/rest/of/path.md)
```

**Reply by feasgal** [reply]
Unfortunately, `mkdocs_macros_plugin` and `pymdownx.snippets` [are not compatible in this way](https://github.com/facelessuser/pymdown-extensions/issues/1948). I've been fighting with the config on both of those half of the day, trying to somehow get their functionality to fire off in the correct order to do what I need, but I've exhausted my knowledge and I'm giving up.

I'll fiddle with your custom hook example and come back here to let the community know if I get it to work. Thank you.

**Reply by squidfunk** [reply]
I was also struggeling with the macros plugin as well – it was impossible to get working for us – and decided to go for [a simple hook](https://github.com/squidfunk/mkdocs-material/pull/5692/files#diff-18481d51124cbd03d2328f15aae5c9276ca9fb5fa98cd4b05b25471c193784e8) (heavily WIP) for our docs rewrite (that's still in progress). We'll be using shortcodes of the following form from now on:

https://github.com/squidfunk/mkdocs-material/blob/45c7e17b06253e96a403da8ec0e935d5b386cc0b/docs/setup/setting-up-the-header.md?plain=1#L52-L54

Maybe that can guide you or other users through the process of how to pull something of like that.

**Comment by feasgal** [comment, marked as answer]
This was solved by MkDocs in version 1.6, with [Validation of absolute links](https://www.mkdocs.org/user-guide/configuration/#validation-of-absolute-links).

```
validation:
  links:
    absolute_links: relative_to_docs
```
